lib/word_formatter.rb

```diff
@@ -0,0 +1,32 @@
+require_relative 'hangman_rules'
+
+class WordFormatter
+
+  attr_reader :rules
+
+  def initialize(rules)
+    @rules = rules
+  end
+
+  def format(secret_word)
+    formatted_word = secret_word.split("").map do |letter|
+      letter = replaced_letter(letter)
```

Comment:
I am pretty sure you don't need this assignment here.
`.map(..)` will create a new array with whatever comes out of the `do |..| .. end` block

Reply:
Even though I'm calling `join()` on it further below?

Reply:
Yeah, you can chain the join() into it.
like this:
```
secret_words.split("").map { |letter| hide_or_reveal(letter) }.join("")
```